### PR TITLE
Fix the flaky test due to m_l_limit_exceeded_exception

### PR DESCRIPTION
**Pull request by zhichao-aws**
### Description
In recent PRs there are flaky test due to m_l_limit_exceeded_exception. 
error log: `"error":{"root_cause":[{"type":"m_l_limit_exceeded_exception","reason":"Memory Circuit Breaker is open, please check your resources!"}],"type":"m_l_limit_exceeded_exception","reason":"Memory Circuit Breaker is open, please check your resources!"},"status":500}` [ref](https://github.com/opensearch-project/skills/actions/runs/7660776972/job/20878791576?pr=148)
This PR fix the flaky test by incrase the threshold of `jvm_heap_memory_threshold`. Add delete model for integ tests if they have deployed one.

Fix the node20 dependency issue by changing codecov to version 1.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by zhichao-aws** [comment]
associate PR: https://github.com/opensearch-project/ml-commons/pull/1943

**Comment by zhichao-aws** [comment]
The CI workflows run 3 times and success all 3 times.

**Review by Hailong-am — APPROVED** [review]

**Review by mingshl — APPROVED** [review]

**Review by chishui — COMMENTED** [review]
nit: better to use `git rebase -i` to reduce commits number and make main branch's commit compact

**Comment by zhichao-aws** [comment]
> nit: better to use `git rebase -i` to reduce commits number and make main branch's commit compact

We will use `squash and merge` to compact the PR to 1 commit then merge to main

**Comment by chishui** [comment]
> We will use `squash and merge` to compact the PR to 1 commit then merge to main

Got it, just feel it's better to reflect the change from the PR and get it reviewed.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,14 +62,14 @@ jobs:
                                ./gradlew publishToMavenLocal"
           
       - name: Upload Coverage Report
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v1
```
Comment (zane-neo) [inline .github/workflows/ci.yml:65]: why do we change the version?

Comment (zhichao-aws) [reply .github/workflows/ci.yml:65]: Our recent CI fails the task of upload coverage report ([ref](https://github.com/opensearch-project/skills/actions/runs/7663590574/job/20886629954)).
Based on the error log:
```
/usr/bin/docker exec  0dc3635fb865fc78d3ab1313d71902228fb8a54822cfa2e7b9cc6ff78f1d73ad sh -c "cat /etc/*release | grep ^ID"
/__e/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.2[7](https://github.com/opensearch-project/skills/actions/runs/7663590574/job/20886629954#step:6:8)' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.2[8](https://github.com/opensearch-project/skills/actions/runs/7663590574/job/20886629954#step:6:9)' not found (required by /__e/node20/bin/node)
/__e/node20/bin/node: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /__e/node20/bin/node)
```
The codecov/codecov-action@v3 use node20 dependency, while we don't have it in linux container. So I changed this to codecov/codecov-action@v1, this is consistent with other plugins.

**Review by zane-neo — APPROVED** [review]

**Pull request merged**